envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -156,24 +158,33 @@ class EnvoySnapshotFactory(
         return newSnapshotForGroup
     }
 
-    private fun getDomainRouteSpecifications(group: Group): Map<DomainRoutesGrouper, Collection<RouteSpecification>> {
+    private fun getDomainRouteSpecifications(
+        group: Group,
+        globalSnapshot: GlobalSnapshot
+    ): Map<DomainRoutesGrouper, Collection<RouteSpecification>> {
         return group.proxySettings.outgoing.getDomainDependencies().groupBy(
             { DomainRoutesGrouper(it.getPort(), it.useSsl()) },
             {
                 RouteSpecification(
                     clusterName = it.getClusterName(),
                     routeDomains = listOf(it.getRouteDomain()),
-                    settings = it.settings
+                    settings = it.settings,
+                    clusterWeights = getTrafficSplittingWeights(group.serviceName, globalSnapshot, it.getClusterName())
                 )
             }
         )
     }
 
-    private fun getDomainPatternRouteSpecifications(group: Group): RouteSpecification {
+    private fun getDomainPatternRouteSpecifications(group: Group, globalSnapshot: GlobalSnapshot): RouteSpecification {
         return RouteSpecification(
             clusterName = properties.dynamicForwardProxy.clusterName,
             routeDomains = group.proxySettings.outgoing.getDomainPatternDependencies().map { it.domainPattern },
-            settings = group.proxySettings.outgoing.defaultServiceSettings
+            settings = group.proxySettings.outgoing.defaultServiceSettings,
+            clusterWeights = getTrafficSplittingWeights(
+                group.serviceName,
+                globalSnapshot,
+                properties.dynamicForwardProxy.clusterName
+            )
```

Comment:
I'm not sure if we should use clusterWeights in domain and domain pattern route specification. Because they are used in domain traffic, which will not be splitted.
@Ferdudas97 Do you have any thoughts about it?

Reply:
removed for this specifications